Assets/MRTK/SDK/Features/Input/Handlers/Constraints/TransformConstraint.cs

```diff
@@ -41,6 +41,27 @@ public ManipulationProximityFlags ProximityType
             set => proximityType = value;
         }
 
+        [SerializeField]
+        [Tooltip("Execution order priority of this constraint. Lower numbers will be executed before higher numbers.")]
+        private int executionOrder = 0;
+
+        /// <summary>
+        /// Execution order priority of this constraint. Lower numbers will be executed before higher numbers.
+        /// </summary>
+        public int ExecutionPriority
+        {
+            get => executionOrder;
+            set {
+                executionOrder = value;
+
+                // Notify all ConstraintManagers to re-sort these priorities.
+                foreach(var mgr in gameObject.GetComponents<ConstraintManager>())
```

Comment:
Would this list be something we could cache in `Awake`, `Start`, or `OnEnable`? Or would we expect managers to be added/removed at runtime?

Also, do we expect there to be multiple managers on one GameObject?

Reply:
Yes, this can be cached. I'll look into this.

Yes, we do expect multiple managers to be present. Not sure why, but there is explicit support for them in our tests and the code.

Reply:
I didn't cache it here because I do not expect users to edit priorities at runtime very often, if at all. I can add it though!

Reply:
> I didn't cache it here because I do not expect users to edit priorities at runtime very often, if at all

That seems reasonable! If this is a super low freq call, we can probably leave it as-is.

That does bring up an additional question: is there a use-case for changing this value at runtime? Or should we just make it an edit-time configuration (like the way Unity handles custom script execution order. I don't believe that can be changed at runtime)

Reply:
I had similar thoughts. But given that the existing constraint system seems geared up to support modifying/altering/doing basically everything with constraints at runtime already, I didn't want to introduce a new mechanism that _didn't_ support runtime editing. I figured that if the entire constraint system was already set up to support runtime editing, I should follow that pattern, because it might have been made that way for a reason.

Reply:
Given the minimal amount of code to support runtime exec order editing, I figured it's a good benefit at very little cost.

Reply:
Oh, here's a use case. Setting up constraints in code. If we made it a readonly value, you couldn't set exec order in code when you're setting up your constraints at runtime.

Reply:
Going to keep this as-is and resolve.

Reply:
Works for me!